### DISCUSSION TITLE
Scale content to fit within a table cell?

### DISCUSSION THREAD

**Discussion by Riboe** [QuestPDF/QuestPDF#122] [Q&A] [answered]
Hi,

I'm generating documents based on some input data from a different system. The data specifies that there must be a series of tables and it also specifies the relative width of the columns in the tables.

The issue, is that I get a `DocumentLayoutException`, due to a column only containing a single character of text and the font size making the character too big to fit in the column.

In an attempt to find all the cases where the text can't fit in the column, I tried to use the `ScaleToFit` element on all the table cells:

    table.Cell().ScaleToFit().Text("text from the data input");

Sadly, this doesn't work. My idea was, that I'd be able to easily see all the cells with noticeably smaller text and then later figure out how to properly size the affected columns.

However, what happens instead, is that the cells instead end up taking up an entire page each.

I'd love some help or pointers to possibly fix my issue.

**Comment by MarcinZiabek** [comment]
Hello 😁 Can you please post a minimal code example that shows the problem? Please also include a screenshot that you get. This should help me finding a solution.

**Reply by Riboe** [reply]
Thanks for the quick reply.

Below is some sample code to reproduce the issue and here's is the output I get: https://imgur.com/a/TisoJwi

I get two pages, each with a single row, however I expected to get a single page which contains both rows. 

```csharp
class Test : IDocument
{
    public DocumentMetadata GetMetadata()
    {
        return DocumentMetadata.Default;
    }

    public void Compose(IDocumentContainer container)
    {
        container.Page(page =>
        {
            page.Margin(40);
            page.DefaultTextStyle(TextStyle.Default.Size(14));

            page.Content().Table(table =>
            {
                table.ColumnsDefinition(colDefs =>
                {
                    colDefs.RelativeColumn(100);
                    colDefs.RelativeColumn(20);
                    colDefs.RelativeColumn(400);
                    colDefs.RelativeColumn(200);
                });

                table.Cell().PaddingHorizontal(4).ScaleToFit().Text("Cell 1");
                table.Cell().PaddingHorizontal(4).ScaleToFit().Text("X");
                table.Cell().PaddingHorizontal(4).ScaleToFit().Text("Cell 3");
                table.Cell().PaddingHorizontal(4).ScaleToFit().Text("Cell 4");

                table.Cell().PaddingHorizontal(4).ScaleToFit().Text("Cell 1");
                table.Cell().PaddingHorizontal(4).ScaleToFit().Text("X");
                table.Cell().PaddingHorizontal(4).ScaleToFit().Text("Cell 3");
                table.Cell().PaddingHorizontal(4).ScaleToFit().Text("Cell 4");
            });
        });
    }
}
```

If you remove the `ScaleToFit()` from the cells containing an `X`, then you'll get a `DocumentLayoutException`, which is to be expected. So that isn't the issue.

**Comment by MarcinZiabek** [comment, marked as answer]
Thank you for your example and finding this issue! Now, I understand the issue. Looks like that I need to refine the `ScaleToFit` implementation to handle cases when one of the dimenstions is not constrained. Current implementation incorrectly assumes that BOTH width and height are limited.

I will do my best to provide fix today/tomorrow.

**Reply by Riboe** [reply]
That sounds great and once again, thanks for the quick replies!

**Comment by MarcinZiabek** [comment]
Hello 😁 I have just published the 2022.2.5 release that fixes this issue. Please check if it solves your problem.

**Reply by Riboe** [reply]
That sounds great! I'll be able to check once I'm back at the office on Monday, so I will let you know then.

**Reply by Riboe** [reply]
Hi,

I finally got some time at work to test this and it's fixed! The new version works as expected, so thanks a ton for that. That'll make our work much easier.

**Reply by MarcinZiabek** [reply]
Thank you for your patience and finding this corner case!

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
